src/models/Deck.ts

```diff
@@ -110,7 +110,16 @@ export class Deck {
     ids?.map((v, i) => (cards[i].noteId = v));
   }
 
-  async pushNewCardsToAnki() {
+  // Calls anki to update the fields of all the passed cards.
+  private async _pushUpdatedCardsToAnki(cards: Card[]) {
+    cards?.map((card) => this.ankiService?.updateFields(card));
+  }
+
+  async createAndUpdateCards() {
+    // Push the updated cards (based on noteID)
+    const updateCards = this.cards.filter((v) => v.noteId);
+    this._pushUpdatedCardsToAnki(updateCards);
```

Comment:
You could probably use reduce so you only cycle through the cards once and return an object maybe {update: [], new: []}```suggestion
    this.cards.reduce((acc, v) => {v.noteId ? acc[update].push(v) : acc[new].push(v) })
```

Or something like that

Reply:
I'll implement a reduce here, or a foreach loop just so we only have to iterate once.